docs/CallflowBuilder/index.md

```diff
@@ -0,0 +1,593 @@
+
+```
+
+###Chaining call flow nodes
+
+Any node can be used as the root element of the call flow. The node invoked first in the node builder chain will be the root element in the callflow.  
+
+```php
+$user_node->addChild($voicemail_node);
+
+```
+The child element is returned after each addChild, so the calls can be chained to build a call flow
+
+```php
+$user_node->addChild($language_node)->addChild($voicemail_node);
+
+```
+
+Subsequent calls to addLastChild will add additional children to the end of the call flow. 
+
+```php
+$user_node->addChild($media_node);
+$user_node->addLastChild($language_node); 
+$user_node->addLastChild($voicemail_node); 
+
+```
+
+addChild() supports an index as a second argument. This is useful for adding menu options. By default the index '_' is used, but can be overridden by specifying a second argument in add child. 
+
+```php
+//sets option one for the menu to $user_node.
+$menu_node->addChild($user_node, 1);
+//sets option two for the menu to $other_user_node
+$menu_node->addChild($other_user_node, 2);
+//sets the default option in the menu to $voicemail_node
+$menu_node->addChild($voicemail_node); 
+
+```
+
+###Building Callflows
+To construct the callflow, build the callflow by invoking the builder build() method, passing the root object used to build the callflow nodes. A full data payload will be constructed which can be assigned to the SDK callflow object via the SDK callflow entitie method fromBuilder(). 
+
+```php
+$your_call_flow = $sdk->Account()->Callflow(); 
+$phone_numbers  = array(1234, 5405551234);
+
+$builder = new Builder($phone_numbers); 
+$data = $builder->build($user_node);
+$your_call_flow->fromBuilder($data);
+$your_call_flow->save(); 
+
+```
+
+###Removing children
+
+Children can be removed from call flows as well by calling the parent object's removeChild() method. The removeChild method will remove the child of the object calling it, preserving rest of the chain by collapsing to remove the child object.  
+
+```php
+$language_node->removeChild(); 
+
+```
+The removeChildren method will remove all the children below the object calling it.
+
+```php
+$user_node->removeChildren(); 
+
+```
+For menus and temporal routes, the removeChild and removeChildren() function supports removal using an optional index argument. 
+
+```php
+//remove the first child of the default option from menu. 
+$menu_node->removeChild('_');
+//remove all the children at option 2 in a menu.
+$menu_node->removeChildren('2');
+//remove the temporal route for index $time_id 
+$temporal_route_node->removeChild($time_id); 
+
+```
+
+##Setting Attributes on callflow nodes
+
+Attributes can be set on the individual node elements depending on the attributes supported by the type of element. For example, the User node's canCallSelf value can be set by calling the User node object's canCallSelf method with the argument TRUE or FALSE. 
+
+```php
+$user_node->canCallSelf(FALSE); 
+
+```
+
+#Callflow nodes
```

Comment:
I think maybe these should be broken out into their own docs, or link to to-be-build kazoo docs.